pina/model/layers/rbf_layer.py

```diff
@@ -0,0 +1,323 @@
+                    device=self.device).float()
+        elif not isinstance(self.smoothing, torch.Tensor):
+            raise ValueError("`smoothing` must be a scalar or a 1-dimensional tensor.")
+
+        self.kernel = self.kernel.lower()
+        if self.kernel not in RADIAL_FUNCS:
+            raise ValueError(f"Unknown kernel: {self.kernel}")
+
+        if self.epsilon is None:
+            if self.kernel in SCALE_INVARIANT:
+                self.epsilon = 1.0
+            else:
+                raise ValueError("Must specify `epsilon` for this kernel.")
+        else:
+            self.epsilon = float(self.epsilon)
+
+        min_degree = MIN_DEGREE.get(self.kernel, -1)
+        if self.degree is None:
+            self.degree = max(min_degree, 0)
+        else:
+            self.degree = int(self.degree)
+            if self.degree < -1:
+                raise ValueError("`degree` must be at least -1.")
+            elif self.degree < min_degree:
+                warnings.warn(
+                    f"`degree` is too small for this kernel. Setting to {min_degree}.",
+                    UserWarning,
+                )
+
+    def fit(self, y, d):
+        self._check_data(y, d)
+        if self.neighbors is None:
+            nobs = self.ny
+        else:
+            raise ValueError("neighbors currently not supported")
+
+        powers = monomial_powers(self.ndim, self.degree).to(device=self.device)
+        if powers.shape[0] > nobs:
+            raise ValueError("The data is not compatible with the requested degree.")
+
+        if self.neighbors is None:
+            self._shift, self._scale, self._coeffs = solve(y, d,
+                    self.smoothing, self.kernel, self.epsilon, powers)
+
+        self.powers = powers
+        self.y = y
+        self.d = d
+
+    def forward(self, x):
+        """
+        Returns the interpolated data at the given points `x`.
+
+        @param x: (n, d) tensor of points at which to query the interpolator
+        @param use_grad (optional): bool, whether to use Torch autograd when
+            querying the interpolator. Default is False.
+
+        Returns a (n, m) tensor of interpolated data.
+        """
+        if x.ndim != 2:
+            raise ValueError("`x` must be a 2-dimensional tensor.")
+
+        nx, ndim = x.shape
+        if ndim != self.y.shape[1]:
+            raise ValueError(
+                "Expected the second dim of `x` to have length "
+                f"{self.y.shape[1]}."
+            )
+
+        kernel_func = RADIAL_FUNCS[self.kernel]
+
+        yeps = self.y * self.epsilon
+        xeps = x * self.epsilon
+        xhat = (x - self._shift) / self._scale
+
+        kv = kernel_vector(xeps, yeps, kernel_func)
+        p = polynomial_matrix(xhat, self.powers)
+        vec = torch.cat([kv, p], dim=1)
+        out = torch.matmul(vec, self._coeffs)
+        out = out.reshape((nx,) + self.d_shape)
+        return out
+
+
```

Comment:
Why the functions below are not members of the class? Should they be static/memeber methods?